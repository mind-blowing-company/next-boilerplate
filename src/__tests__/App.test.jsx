import React from "react";
import renderer from "react-test-renderer";
import {Provider} from "react-redux";
import {mount} from "enzyme";

import App from "../pages/App";
import initStore from "../redux/store";

jest.mock("../i18n", () => ({
    withTranslation: () => (component) => component
}));

test("works with snapshot", () => {
    const component = renderer.create(
        <Provider store={initStore()}>
            <App i18n={{language: "en_US"}} t={key => key}/>
        </Provider>
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
});

test("works with Enzyme", () => {
    const page = mount(
        <Provider store={initStore()}>
            <App i18n={{language: "en_US"}} t={key => key}/>
        </Provider>
    );

    expect(page.find("button").text()).toEqual("switch-locale");
});
