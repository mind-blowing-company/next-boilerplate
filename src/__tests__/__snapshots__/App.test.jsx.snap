// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`works with snapshot 1`] = `
<div
  className="app"
>
  <div
    className="app-container"
  >
    <button
      className="switch-locale"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      switch-locale
    </button>
    <img
      alt="logo"
      className="app-logo"
      src="/logo.svg"
    />
    <p>
      edit
       
      <code>
        pages/index.js
      </code>
       
      save-reload
      .
    </p>
    <p>
      06:00:00
    </p>
    <a
      className="app-link"
      href="https://nextjs.org"
      rel="noopener noreferrer"
      target="_blank"
    >
      learn
    </a>
  </div>
</div>
`;
